File: 0944. Delete Columns to Make Sorted/Solution0944.go
package main

import (
	"fmt"
)

func main() {
	strs := []string{"cba", "daf", "ghi"}
	res := minDeletionSize(strs)
	fmt.Println(res)
}

func minDeletionSize(strs []string) int {
	row, col := len(strs), len(strs[0])
	res := 0
	for c := 0; c < col; c++ {
		for r := 1; r < row; r++ {
			if strs[r][c] < strs[r-1][c] {
				res++
				break
			}
		}
	}
	return res
}

File: 0944. Delete Columns to Make Sorted/Solution0944.js
/**
 * @param {string[]} strs
 * @return {number}
 */
var minDeletionSize = function(strs) {
    let row = strs.length, col = strs[0].length;
    let res = 0;
    for (let i = 0; i < col; i++) {
        for (let j = 1; j < row; j++) {
            if (strs[j][i] < strs[j-1][i]) {
                res++;
                break;
            }
        }
    }
    return res;     
};

var strs = ["cba","daf","ghi"];
var res = minDeletionSize(strs);
console.log(res)